src/main/java/io/jenkins/plugins/fontawesome/api/FontAwesomeIcons.java

```diff
@@ -0,0 +1,112 @@
+package io.jenkins.plugins.fontawesome.api;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.*;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Utility to work with icons provided by the font-awesome-api-plugin.
+ *
+ * @author strangelookingnerd
+ */
+public final class FontAwesomeIcons {
+
+    private static final Logger LOGGER = Logger.getLogger(FontAwesomeIcons.class.getName());
+    private static final String SVG_FILE_ENDING = ".svg";
+    private static final String IMAGES_SYMBOLS_PATH = "images/symbols/";
+    private static final String FONT_AWESOME_API_PLUGIN = "font-awesome-api";
+    private static final String ICON_CLASS_NAME_PATTERN = "symbol-%s plugin-" + FONT_AWESOME_API_PLUGIN;
+
+    private static final FontAwesomeIcons INSTANCE = new FontAwesomeIcons();
+
+    private final Map<String, String> availableIcons = new ConcurrentSkipListMap<>();
+
+    private FontAwesomeIcons() {
+        try {
+            Enumeration<URL> urls = getClass().getClassLoader().getResources(IMAGES_SYMBOLS_PATH);
+
+            while (urls.hasMoreElements()) {
+                URL url = urls.nextElement();
+
+                if (StringUtils.contains(url.toExternalForm(), FONT_AWESOME_API_PLUGIN)) {
+                    URI uri = url.toURI();
+
+                    if (StringUtils.equals(uri.getScheme(), "jar")) {
+                        try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.emptyMap())) {
+                            collectIcons(fileSystem.getPath(IMAGES_SYMBOLS_PATH));
+                        }
+                    }
+                    else {
+                        collectIcons(Paths.get(uri));
+                    }
+                    break;
+                }
```

Comment:
Do both branches make sense in production environment?

Reply:
Not really. It is mainly for (integration) testing when we need to use the "exploded" jar file.